app/models/user.rb

```diff
@@ -823,14 +822,11 @@ def check_login
     end
 
     # check if login already exists
-    base_login = login.downcase.strip
+    exists = User.find_by(login: login)
+    return true if !exists || exists.id == id
 
-    alternatives = [nil] + Array(1..20) + [ SecureRandom.uuid ]
-    alternatives.each do |suffix|
-      self.login = "#{base_login}#{suffix}"
-      exists = User.find_by(login: login)
-      return true if !exists || exists.id == id
-    end
+    # we leave the email uniqueness check for ensure_uniq_email
+    return true if email.present?
```

Comment:
this is needed since other parts of code expect activerecord errors when uniqueness check fails (e.g. email)